tools/check_all_packages.jl

```diff
@@ -0,0 +1,64 @@
+# hacky script to parse all Julia files in all packages in General
+# to Exprs and report errors
+#
+# Run this after registry_download.jl (so the pkgs directory is populated).
+
+using JuliaSyntax, Logging
+
+logio = open(joinpath(@__DIR__, "logs.txt"), "w")
+logger = Logging.ConsoleLogger(logio)
+
+pkgspath = joinpath(@__DIR__, "pkgs")
+
+parallel = 50
+exceptions = []
+Logging.with_logger(logger) do
+    for tars in Iterators.partition(readdir(pkgspath), parallel)
+        @sync for tar in tars
```

Comment:
Is this partition essentially doing the same as `asyncmap()` but manually?

Reply:
Yup, looks like it :)